### ISSUE TITLE
Garmin Fenix 6 dont detecte QZ as a sensor[BUG]

### ISSUE THREAD

**Issue by PieroPatagonia** [cagnulein/qdomyos-zwift#2285]
**Describe the bug**
The Fenix 6 doesnt detectet the QZ as a sensor

**To Reproduce**

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: VIVO V25
 - OS: Funtouch OS 14

**Append a debug log**
[debug-Tue_Apr_9_14_14_42_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14922841/debug-Tue_Apr_9_14_14_42_2024.log)

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**

**Comment by cagnulein** [comment]
Hi @PieroPatagonia in order to address correctly the issue I need a bug report from android. Are you able to do so?

https://developer.android.com/studio/debug/bug-report?hl=it

**Comment by PieroPatagonia** [comment]
Is it the attached file?

**Comment by cagnulein** [comment]
The one that you uploaded before it was a qz debug log. I need the android
bug report. if you just attached a new file you have to do on github
otherwise i can't see it

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 9 apr 2024 alle 21:04 PieroPatagonia ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
yes but you have to wait the finish of the upload before pushing comment otherwise the file is corrupted

**Comment by PieroPatagonia** [comment]
[bugreport-V2202-UP1A.231005.007_SC-2024-04-09-15-11-17.zip](https://github.com/cagnulein/qdomyos-zwift/files/14922986/bugreport-V2202-UP1A.231005.007_SC-2024-04-09-15-11-17.zip)

**Comment by cagnulein** [comment]
thanks, i will check it tomorrow

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 9 apr 2024 alle 21:18 PieroPatagonia ***@***.***>
ha scritto:

**Comment by PieroPatagonia** [comment]
Ok Roberto, thank for your patience jajaja

**Comment by cagnulein** [comment]
hi @PieroPatagonia i checked the bugreport but i don't see qz running in this, did you collect the bug report after trying to connect the watch to qz? i have to see the error in the log.
Thanks

**Comment by PieroPatagonia** [comment]
Hi Roberto, i will do it tomorrow, with the TDF and Garmin.
Thanks

**Comment by PieroPatagonia** [comment]
GHi Roberto, this is the report tryaing to connect QZ to Garmin
[bugreport-V2202-UP1A.231005.007_SC-2024-04-11-14-06-39.zip](https://github.com/cagnulein/qdomyos-zwift/files/14949861/bugreport-V2202-UP1A.231005.007_SC-2024-04-11-14-06-39.zip)
 TDF connected

**Comment by cagnulein** [comment]
Thanks I will check it tomorrow

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 11 apr 2024 alle 20:15 PieroPatagonia <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
that's strange @PieroPatagonia 
in the log everything is working properly and I also see the answers from the watch too! So I guess something in the watch setup that's not right. Maybe can you make me a video of what you are trying to do on the watch side? anyway from the software and hardware side on the phone everything is working

**Comment by PieroPatagonia** [comment]
Ok Roberto, I will do it and send you the video. Thanks again.

**Comment by PieroPatagonia** [comment]
Thanks Roberto

https://github.com/cagnulein/qdomyos-zwift/assets/143635813/2617ef98-0334-4fb7-b2d2-97535e70444e

**Comment by cagnulein** [comment]
that's so strange, on sunday i will retest it myself and i will let you know to be sure

**Comment by cagnulein** [comment]
just tried and it works at first shot https://github.com/cagnulein/qdomyos-zwift/assets/48216/ff645e63-ba81-4c50-9531-5cfb0ef983e2

i will re examine your log next week, maybe i didn't see an issue

**Comment by cagnulein** [comment]
did you try to look also for power sensor on the watch? the watch seems to go very fast on timeout, my garmin edge is stay there at least for 20 seconds

**Comment by PieroPatagonia** [comment]
Hello Roberto, yes I did it and it doesn't recognize it.
To test if it was a problem with the watch, I connected the power
sensor of my bicycle to the Watch and it recognized it immediately.

I was also able to check that the phone captures the bicycle's power sensor.
I don't know how to check if the phone sends the QZ information through Ant+.

**Comment by cagnulein** [comment]
i guess i have to buy a used watch too to test it. i will try to find good
offers and i will let you know. someone near to you has a garmin edge like
mine so you can do the same test?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 14 apr 2024 alle 15:56 PieroPatagonia <
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
found and bought a fenix 5

i will let you know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 14 apr 2024 alle 15:58 cagnulein ***@***.***> ha
scritto:

**Comment by PieroPatagonia** [comment]
I have other Cycle computers from other brands that recognize the
power sensor on my bike and I will try it.

I'll ask my friends if they have any Edge to try it too.

Thanks Roberto.

**Comment by cagnulein** [comment]
thanks! let's fight these issues from every field :)

**Comment by PieroPatagonia** [comment]
Sure! 💪

**Comment by PieroPatagonia** [comment]
Hi Roberto. I tried to connect with a Igpsport odómetro and dont reconise
any sensor.

It is possible run Qz o a computer? And test if work on it.

**Comment by PieroPatagonia** [comment]
May be if you let me know wich app are you using for the USB ant+ on the
phone. And check that.

**Comment by cagnulein** [comment]
qz uses ant+ only on android . i don't understand your last question. what
do you mean which app?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 14 apr 2024 alle 18:03 PieroPatagonia <
***@***.***> ha scritto:

**Comment by PieroPatagonia** [comment]
From the Playstore a instaled these
![Screenshot_2024_0414_121523](https://github.com/cagnulein/qdomyos-zwift/assets/143635813/dd287ca8-1311-4082-b1d1-d8132e2d6d88)
 apps  on the phone to make the usb dongle run. 
Maybe if i install other or unistall somo of that will work

**Comment by cagnulein** [comment]
yes they are right. infact i don't see error in your bug report. it seems an issue of the watch. let's see what we will find out

**Comment by cagnulein** [comment]
a fenix 5 should arrive to me next week (40€ :) )

**Comment by PieroPatagonia** [comment]
Great !!! Thanks Roberto

**Comment by PieroPatagonia** [comment]
Hi Roberto. I have some news, i instale QZ on an old Galaxy Tablet and works, without de usb dongle. So now i can see the metrics on the fenix 6.

Why didnt work in my vivo phone, i dont know jjaja.

**Comment by cagnulein** [comment]
> Hi Roberto. I have some news, i instale QZ on an old Galaxy Tablet and works, without de usb dongle. So now i can see the metrics on the fenix 6.
> 
> Why didnt work in my vivo phone, i dont know jjaja.

so that's about the dongle! so the garmin one is not compatible?! :O 



> Hi Roberto, me again the one problem i dtected in the data was the speed, i belive i have to put some diameter of the well? I wish to do more tha 100 km / hr jjajaa

you have to put wheel diameter to 100mm

let me know

**Comment by cagnulein** [comment]
@PieroPatagonia are we OK with this?

**Comment by PieroPatagonia** [comment]
Hi Roberto, yes is working using the galaxy tablet. Just a detail sometime on the garmin fénix 6, the conexión break with the power and cadence speed sensor. After a few second reconect again. Is just a detail.

**Comment by cagnulein** [comment]
How often?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 13 mag 2024 alle 21:13 PieroPatagonia <
***@***.***> ha scritto:

**Comment by PieroPatagonia** [comment]
It is variable, there are very stable moments and on other occasions every couple of minutes or even more often.

**Comment by cagnulein** [comment]
With the ant plugins app from another device with ant are you able to get
the ant signal from qz? If so are you seeing the dropouts also there?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 13 mag 2024 alle 21:54 PieroPatagonia <
***@***.***> ha scritto:

**Comment by PieroPatagonia** [comment]
I haven't tried, I'm going to try with my phone and the ant+ dungle and I'll let you know.

**Comment by cagnulein** [comment]
Ok, sounds good.


Il giorno lun 13 mag 2024 alle 22:06 PieroPatagonia <
***@***.***> ha scritto:

**Comment by PieroPatagonia** [comment]
Hi Roberto, i tested with igsport bike computer and the Fenix 6 at the same time.  Like 
 I told you before works fine but in both drop the signal for a few second in diferent moments. In a Sesion of one hours normaly drops five time.

**Comment by cagnulein** [comment]
Do you have only one phone with ant+? I would like to check if it's a phone
issue
Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 16 mag 2024 alle 02:31 PieroPatagonia <
***@***.***> ha scritto:

**Comment by PieroPatagonia** [comment]
Yes, I can see it. But some time drops the data of speed and continue the
speed and cadence data.

**Comment by oel59** [comment]
Thanks!
I have solved my problem in the meantime.

**Comment by cagnulein** [comment]
@oel59 do you see some dropouts too?

**Comment by oel59** [comment]
I have not noted any dropouts, but I have not paid too much attention to that aspect. I will have a close look at my next ride.

**Comment by cagnulein** [comment]
Ok, let me know.


Il giorno lun 20 mag 2024 alle 19:43 oel59 ***@***.***> ha
scritto:

**Comment by oel59** [comment]
Hi, I have done several rides with the sensors connected to my Garmin watch, and I have not noticed any dropouts.

**Comment by cagnulein** [comment]
That's great @oel59 !

**Comment by PieroPatagonia** [comment]
Hi Roberto, just find out, that my drops problems, is a Samsung Tab
problem. You can see it on this link

https://eu.community.samsung.com/t5/tablets/ant-bluetooth-amp-flight-mode/td-p/4618498

**Comment by cagnulein** [comment]
Maybe you can buy an ant dongle to fix it

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 1 giu 2024 alle 17:10 PieroPatagonia ***@***.***>
ha scritto:

**Comment by PieroPatagonia** [comment]
Ok Finally i solve the problem, just download an update of the samsung tab
s2 Ant+ driver. Doesnt do it automaticly. It is necesary go to the play
store and look for Ant radio service and frome there update.
Now is working fine.
Thanks Roberto for all the help.

**Comment by cagnulein** [comment]
Glad it worked!

Il giorno lun 24 giu 2024 alle 03:52 PieroPatagonia <
***@***.***> ha scritto:
